index.html

```diff
@@ -857,6 +852,91 @@
+\end{equation}
+where $\Delta(E_b, \eta_c)$ is a darkening factor given by
+\begin{equation} \label{general_darkening_formula}
+\Delta(E_b, \eta_c) = \frac{1 - K}{1 - E_b K} \ .
+\end{equation}
+We can thus approximate the required boost factor as the reciprocal of this, i.e. $B_0 \approx \Delta^{-1}$. Note that as $E_b \rightarrow 1$, the darkening factor $\Delta \rightarrow 1$, as required since physically no energy is lost if the base reflects all incident energy (in the clear-coat case). If the base albedo $E_b$ is boosted by $B(\delta)$, then clearly at $\delta=0$ the darkening factor cancels and equation [undarkened_coat_albedo] is satisfied as required.
+
+Here $K \in [0,1]$ is the _internal diffuse reflection coefficient_, which corresponds to the fraction of the energy leaving the base which returns
+to the base due to internal reflection from the coat. Higher values of $K$ lead to more internal reflections, thus more darkening and saturation. In the case of a Lambertian base (which should be a reasonable approximation to the rough metal, dielectric, or diffuse cases), $K = K_r$ where, with relative coat IOR $\eta_c$:
+\begin{equation} \label{internal_diffuse_reflection_coefficient_for_rough_base}
+K_r = 1 - \Bigl(1 - E_F(\eta_c)\Bigr)/\eta_c^2 \ .
+\end{equation}
+The albedo $E_F(\eta_c)$ is the _hemispherical_ (or _average_) albedo of the coat Fresnel factor [^avg_fresnel]. If the base has specular reflection (due to a smooth metallic or dielectric interface) then equation [general_darkening_formula] still applies, but the appropriate value of the internal diffuse reflection coefficient $K$ should be closer to:
+\begin{equation} \label{internal_diffuse_reflection_coefficient_for_smooth_base}
+K_s = F(\omega_o, \eta_c) \ ,
+\end{equation}
+which becomes exact in the limit of a perfectly smooth metallic or dielectric base. There is significantly less darkening in the case of a smooth base, as the specular reflection from the base is less likely to generate total internal reflection at the coat boundary.
+
+We thus recommend in the general case that the darkening factor of equation [general_darkening_formula] be applied, but taking
+\begin{equation} \label{internal_diffuse_reflection_coefficient_for_general_base}
+K = \mathrm{lerp}(K_s, K_r, r_b)
+\end{equation}
+where $r_b$ is an estimate of the base roughness, a blend between metal $r_m$ and dielectric $r_d$ roughness estimates according to the **`base_metalness`** $\mathtt{M}$:
+\begin{equation} \label{base_roughness_estimate}
+r_b = \mathrm{lerp}(r_d, r_m, \mathtt{M}) \ .
+\end{equation}
+The base dielectric roughness $r_d$ can reasonably be estimated as a mix between the high roughness of an assumed underlying base and the microfacet dielectric roughness $r$ = **`specular_roughness`**, according to the base dielectric Fresnel factor modulated via **`specular_weight`** $\xi_s$ (see equation [unmodulated_fresnel] and equation [modulated_ior]):
```

Comment:
I'm assuming this Fresnel factor uses the relative IOR of the base and the coat (and not just the base IOR). Is that clear here? Or would that make things more complex (e.g., due to the possibility or TIR arising here)?

Reply:
It should be clarified that by Fresnel factor here I meant specifically the F0 from equation 25.

I'm not sure it's clear, but the Fresnel factor (of equation 25) is supposed to take into account the presence of the coat (so e.g. goes to zero if the coat and base dielectric have equal IOR). 

Yes TIR could occur, if the coat IOR > base IOR. In which case this Fresnel factor goes to 1, so we would take the base roughness to be `specular_roughness` in computing the darkening factor.

The idea of using the Fresnel factor in the lerp (for the dielectric base roughness estimate) was just that the dielectric specular reflection will be superimposed on some diffuse reflection, and the F0 seems a reasonable proxy for how much we should weight the dielectric roughness in the blend. Obviously, this is just a rough heuristic, attempting to capture the essence of the effect (that a smooth base darkens less) in as simple as possible a way.